web/src/main/resources/static/js/tree.js

```diff
@@ -592,7 +598,8 @@ var tree = (function () {
             if (searchKey === '') {
                 d3.select(this).style('fill', 'black');
             } else {
-                if (d.name.toLowerCase().indexOf(searchKey) !== -1) {
+                if (d.name.toLowerCase().indexOf(searchKey) !== -1 ||
+                    (typeof d.history !== 'undefined' && d.history != '' && d.history.toLowerCase().split(",").indexOf(searchKey) !== -1)) {
```

Comment:
very nice .. enable searching / highlighting based on the previous codes that have been loaded.
The whole search function could be made more efficient .. as a later enhancement.